resources/sets/base_set/pieces/pawn.py

```diff
@@ -0,0 +1,156 @@
+        """
+        moves = []
+
+        for direction in self.DIRECTIONS:
+            # Consider open spaces
+            # Check space 1 in front
+            move_f = self._current_coords + direction
+            if self._board.query_space(move_f) is None:
+                moves.append(move_f)
+
+            # Check space 2 in front (if in starting spot)
+            move_ff = self._current_coords + (2 * direction)
+            if (
+                    self._current_coords == self._starting_coords
+                    and self._board.query_space(move_ff) is None
+            ):
+                moves.append(move_ff)
+
+            # Consider captures
+            move_fl = self._current_coords + direction + np.array([1, 0])
+            player_fl = self._board.query_space(move_fl)
+            if (
+                    player_fl is not None
+                    and player_fl.player_controller != self._player_controller
+            ):
+                moves.append(move_fl)
+
+            move_fr = self._current_coords + direction + np.array([-1, 0])
+            player_fr = self._board.query_space(move_fr)
+            if (
+                    player_fr is not None
+                    and player_fr.player_controller != self._player_controller
+            ):
+                moves.append(move_fr)
+
+        return moves
+
+    def _get_promotion_zone(self) -> int:
+        """
+        Get the y-value associated with the pawn "promotion zone" in the pawn's
+            forward direction.
+
+        :return: y-value at which the pawn will promote
+        :rtype: int
+        """
+        if self._moving_down():
+            return 0
+        else:
+            return cast(int, self._board.dimensions[1]) - 1
+
+    def move(self, coords: ndarray[int]) -> bool:
+        """
+        Performs an action on the piece using arguments.
+
+        :param coords: Coordinates of the attempted move
+        :type coords: ndarray[int]
+        :return: True on successful move, False otherwise.
+        :rtype: bool
+        """
+
+        # Check if move is valid
+        if coords not in self.list_moves():
+            print("Invalid Move.")
+            return False
+
+        other_piece = self._board.query_space(coords)
+
+        # Check for capture
+        if (
+                other_piece is not None
+                and other_piece.player_controller != self._player_controller
+        ):
+            other_piece.capture(self._player_controller)
+
+        # Check for promote
+        promotion_zone = self._get_promotion_zone()
+        if coords[1] == promotion_zone:
+            # TODO request promotion input from user
+            self._board.add_piece(Queen(
+                self._board,
+                self._piece_char,
+                self._piece_pxl_hex,
+                self._current_coords,
+                self._player_controller
+            ))
+
+        self._current_coords = coords
```

Comment:
This does not update the board's layout with the updated coordinates. Add something like
```
self._board.remove_piece(self)
self._current_coords = coords
self._board.add_piece(self)
```